### PR TITLE
Move to PHPUnit 6

### PR DESCRIPTION
**Pull request by Slamdunk**
- [x] Migrate to PHPUnit 6 (thanks to [PHP-CS-Fixer](https://github.com/FriendsOfPHP/PHP-CS-Fixer))
- [x] Drop `composer.lock`, so drop Travis `DEPS=locked`
- [x] Align dependencies to https://github.com/doctrine/DoctrineModule/pull/613
- [x] Test against PHP 7.2 and pass
- [x] Strip Travis duplicated builds
- [x] Make `CliController` output mockable

Supersedes https://github.com/doctrine/DoctrineModule/pull/610 and https://github.com/doctrine/DoctrineModule/pull/613

### CODE REVIEW

File: tests/DoctrineModuleTest/Component/Console/Input/RequestInputTest.php [hunk 1]
```diff
@@ -18,7 +18,7 @@
 
 namespace DoctrineModuleTest\Component\Console\Input;
 
-use PHPUnit_Framework_TestCase as TestCase;
+use PHPUnit\Framework\TestCase as TestCase;
```
Comment (Majkl578) [inline tests/DoctrineModuleTest/Component/Console/Input/RequestInputTest.php]: The alias is useless.

File: tests/DoctrineModuleTest/Component/Console/Output/PropertyOutputTest.php [hunk 2]
```diff
@@ -18,7 +18,7 @@
 
 namespace DoctrineModuleTest\Component\Console\Output;
 
-use PHPUnit_Framework_TestCase as TestCase;
+use PHPUnit\Framework\TestCase as TestCase;
```
Comment (Majkl578) [inline tests/DoctrineModuleTest/Component/Console/Output/PropertyOutputTest.php]: The alias is useless.

File: tests/DoctrineModuleTest/Form/Element/ProxyAwareElementTestCase.php [hunk 3]
```diff
@@ -21,7 +21,7 @@
 
 use Doctrine\Common\Collections\ArrayCollection;
 use DoctrineModuleTest\Form\Element\TestAsset\FormObject;
-use PHPUnit_Framework_TestCase as TestCase;
+use PHPUnit\Framework\TestCase as TestCase;
```
Comment (Majkl578) [inline tests/DoctrineModuleTest/Form/Element/ProxyAwareElementTestCase.php]: The alias is useless.

File: tests/DoctrineModuleTest/Validator/Service/NoObjectExistsFactoryTest.php [hunk 4]
```diff
@@ -2,7 +2,7 @@
 
 namespace DoctrineModule\Validator\Service;
 
-use PHPUnit_Framework_TestCase as TestCase;
+use PHPUnit\Framework\TestCase as TestCase;
```
Comment (Majkl578) [inline tests/DoctrineModuleTest/Validator/Service/NoObjectExistsFactoryTest.php]: The alias is useless.

File: tests/DoctrineModuleTest/Validator/Service/ObjectExistsFactoryTest.php [hunk 5]
```diff
@@ -2,7 +2,7 @@
 
 namespace DoctrineModule\Validator\Service;
 
-use PHPUnit_Framework_TestCase as TestCase;
+use PHPUnit\Framework\TestCase as TestCase;
```
Comment (Majkl578) [inline tests/DoctrineModuleTest/Validator/Service/ObjectExistsFactoryTest.php]: The alias is useless.

File: tests/DoctrineModuleTest/Validator/Service/UniqueObjectFactoryTest.php [hunk 6]
```diff
@@ -2,7 +2,7 @@
 
 namespace DoctrineModule\Validator\Service;
 
-use PHPUnit_Framework_TestCase as TestCase;
+use PHPUnit\Framework\TestCase as TestCase;
```
Comment (Majkl578) [inline tests/DoctrineModuleTest/Validator/Service/UniqueObjectFactoryTest.php]: The alias is useless.

**Comment by Slamdunk** [comment]
1. Useless aliases dropped
1. nightly build on latest deps reintroduced

> I don't really approve the changes you made to CI build matrix. If you want to refactor Travis build, please do so using Build Stages as we did in some repos (i. e DBAL, ORM) already - it's more readable and understandable.

I didn't refactored it, I just dropped duplicated builds, nightly and locked ones. Now nightly are reintroduced on latest.
I've done it so to ease the mergeability.

I am able to move travis to stages, but this repo seems a bit stalled, I don't want to create a huge PR that will take a lot of time to be understood and merged.

> I don't understand why you removed nightly, it's perfectly legit to test against php-next (7.3 now) to spot possible issues early. The fact it's failing now for master with PHP fatal error is not justification to remove it, it should be fixed if still relevant (maybe fixed by deps bump, else separately from this PR).

I've not dropped it because it failing: as you can see with my PR it passes without specific changes.

I've dropped it because at the time of writing there is no difference between 7.2 and nightly.
A new release is a year apart from now and we have no clue if the changes made in the next months will see the light or not. To me **at the time of writing** nightly build are just a waste of computational resources, and I'm trying to be as much environmentalist as I can.

**Comment by Majkl578** [comment]
> I've not dropped it because it was failing: as you can see with my PR it passes without specific changes.

Ok, coincidence - if you look at master, there are three failures (two different issues) on nightly: https://travis-ci.org/doctrine/DoctrineModule/builds/309067889

Looks like one failure was caused by old PHPUnit: https://travis-ci.org/doctrine/DoctrineModule/jobs/309067901
But not this one, with lowest deps on nightly: https://travis-ci.org/doctrine/DoctrineModule/jobs/309067899

> I am able to move travis to stages, but this repo seems a bit stalled, I don't want to create a huge PR that will take a lot of time to be understood and merged.

Fine, leave it as is. :)

**Comment by TomHAnderson** [comment]
@Slamdunk this module is stalled.  I agree with your choice to move forward without stages and implement those once DoctrineModule is up to speed with PHP 7.1

**Comment by Slamdunk** [comment]
> this module is stalled

Does this mean that the project will still be maintained with a PR such this or not?

I run a million euro business on DoctrineModule + DoctrineORMModule: I am intended to keep them functioning and up to date, help with PR like this.

**Comment by Ocramius** [comment]
Yeah, it will be maintained, just not improved for now.

**Comment by TomHAnderson** [comment]
@Slamdunk I like this PR and I'll merge it into dev-master soon.  I don't have a timeline for a release at this time though.

**Comment by Majkl578** [comment]
Since PHPUnit 7 is out, would probably make sense to upgrade here too?

**Comment by Slamdunk** [comment]
@Majkl578 Done :+1:

**Comment by Slamdunk** [comment]
I've added 7.0 to composer, but zend-test disallow it

**Comment by TomHAnderson** [comment]
ok, thanks for trying that @Slamdunk 

I'll dig into this PR tomorrow.

**Comment by TomHAnderson** [comment]
Released with pre-release 2.0.1

**Pull request merged**